lib/Messenger/Handler/AssetUpdateTasksHandler.php

```diff
@@ -62,8 +63,18 @@ private function saveAsset(Asset $asset): void
 
     private function processDocument(Asset\Document $asset): void
     {
-        if ($asset->getMimeType() === 'application/pdf' && $asset->checkIfPdfContainsJS()) {
-            $asset->save(['versionNote' => 'PDF scan result']);
+        if ($asset->isPdfScanningEnabled() && $asset->getMimeType() === 'application/pdf' && !$asset->getScanStatus()) {
+            $asset->setCustomSetting($asset::CUSTOM_SETTING_PDF_SCAN_STATUS, PdfScanStatus::IN_PROGRESS->value);
+            $this->saveAsset($asset);
+
+            if ($asset->checkIfPdfContainsJS()) {
+                $asset->setCustomSetting($asset::CUSTOM_SETTING_PDF_SCAN_STATUS, PdfScanStatus::SAFE->value);
+                $note = 'safe';
+            } else {
+                $asset->setCustomSetting($asset::CUSTOM_SETTING_PDF_SCAN_STATUS, PdfScanStatus::UNSAFE->value);
+                $note = 'unsafe';
+            }
+            $asset->save(['versionNote' => 'PDF scan result:' . $note]);
```

Comment:
mhhh, this could be bad, if upload a pdf that takes 5 seconds to read, and within there's a new one that takes 1 second, by the time it finished the first task, it would replace the latest or second one.
That's probably why it uses saveAsset() that avoid Versioning.
Is it necessary vernsioNote? could it be moved to metadata?

Reply:
I did all the changes in all 3 files, but it's still not working for me in Pimcore 11.2

Reply:
Same , @kingjia90 I did all changes but same  screen appear.